rust/src/format/manifest.rs

```diff
@@ -77,6 +77,24 @@ impl Manifest {
             Utc,
         )
     }
+
+    pub fn max_fragment_id(&self) -> Option<u64> {
+        self.fragments.iter().map(|f| f.id).max()
+    }
+
+    pub fn fragments_since(&self, since: Manifest) -> Vec<Fragment> {
+        let mut fragments = vec![];
+        let mut fragment_map = HashMap::new();
```

Comment:
should we check `since` version is ealier than `self.version`? 

Feel it is more reliable that fragment has a "version" attach to it, so that this function just compares the version number.

To handle case like, deletion / insert and etc

Reply:
i can add an assert check for the version.